extensions/pkg/apis/config/types.go

```diff
@@ -23,4 +27,17 @@ type HealthCheckConfig struct {
 	// already running on them).
 	// defaults to 30 sec
 	SyncPeriod metav1.Duration
+	// ShootRESTOptions allow overwriting certain default settings of the shoot rest.Config.
+	ShootRESTOptions RESTOptions
```

Comment:
Shouldn't this be a pointer?

Reply:
It is similar to `SyncPeriod` which is optional too, but not a pointer.

Reply:
Newly introduced field should be a pointer. @ialidzhikov Can you please confirm here?

Reply:
Hm, this rule applies for exposed APIs by the gardener-apiserver such as Shoot, etc. I am not sure whether we strictly follow it for component config APIs. 
For such component configuration APIs [./docs/development/changing-the-api.md](https://github.com/gardener/gardener/blob/v1.51.0/docs/development/changing-the-api.md#component-configuration-apis) writes that:
> Hence, the above checklist also applies for changes to those APIs.
> However, since these APIs are only used internally and only during the deployment of Gardener the guidelines with respect to changes and backwards-compatibility are slightly relaxed.

Reply:
It's probably still better to follow these guidelines even here (why would we not do it)? Not sure why it wasn't done for `SyncPeriod`, but let's at least do it for newer fields.

Reply:
`ShootRESTOptions` is a pointer now.
Additionally, I removed my extension of `AddArgs` structure, because it is not necessary.